arangod/Aql/Functions.cpp

```diff
@@ -7567,10 +7567,22 @@ AqlValue Functions::SchemaValidate(ExpressionContext* expressionContext,
         TRI_ERROR_BAD_PARAMETER, "second parameter is not a schema object: " +
                                      schemaValue.slice().toJson());
   }
+  auto* validator = expressionContext->buildValidator(schemaValue.slice());
+
+  //store and restore validation level this is cheaper than modifying the VPack
+  auto storedLevel = validator->level();
 
-  arangodb::ValidatorJsonSchema validator(schemaValue.slice());
-  validator.setLevel(ValidationLevel::Strict); //override level so the validation will be executed no matter what
-  auto res = validator.validateOne(docValue.slice(), vpackOptions);
+  //override level so the validation will be executed no matter what
+  validator->setLevel(ValidationLevel::Strict);
+
+  Result res;
+  {
+    arangodb::ScopeGuard guardi([storedLevel, &validator]{
+        validator->setLevel(storedLevel);
+    });
+
+    res = validator->validateOne(docValue.slice(), vpackOptions);
+  }
```

Comment:
Wouldn't it be easier to build all the schemas with validation level `Strict` from the very beginning?

Reply:
This would us require to rewrite the Slice/AQLValue. which is certainly more expensive. Alternatively we could use use a different constructor. This would only slightly clutter the Validator creation. But then we would store a Validator that is different to the description that
we hash and then use as the key in the `_validatorCache` map. I am not sure If this should be done.